global-state/jar/src/com/sixsq/slipstream/globalstate.clj

```diff
@@ -0,0 +1,83 @@
+(ns com.sixsq.slipstream.globalstate
+  "SlipStream  global state project."
+  (:require
+    [clojure.tools.logging :as log]
+    [clj-time.core :as t]
+    [qbits.spandex :as spandex]
+    [clojure.core.async :as async]
+    [environ.core :as env]
+    [com.sixsq.slipstream.scheduler :as scheduler]))
+
+
+;;define the target index for the global state
+(def ^:const index-action {:index {:_index "resources-index" :_type "global-state-snapshot"}})
+;;initial delay
+(def ^:const immediately 0)
+;;task interval
+(def ^:const every-minute (* 60 1000))
+;;global ES query to search every virtual-machine documents
+(def ^:const search-all-virtual-machines-url "resources-index/virtual-machine/_search")
```

Comment:
Probably also better to parameterize this as well.  If we end up with storage or other resources to meter, do you think that we'll have a separate service for each or will extend this one to include those other resources?

Reply:
My personal preference would be to extend this metering service so to include any other resources (like storage) in the future where this process would loop over a defined set of resources to meter (virtual machine, storage...). However my first attempt to do so, probably because of the current implementation of the scheduler

Reply:
Created 2 env variables for parameterizing : 
METERING_INTERVAL=60000
METERING_SEARCH_URL=resources-index/virtual-machine/_search